### DISCUSSION TITLE
Use of transform and/or render package for cloud-optimised geotiff tiles?

### DISCUSSION THREAD

**Discussion by rowanwins** [allmaps/allmaps#6] [General ideas and discussions]
Hi there,

I've been doing some work at consuming cloud optimised geotiffs (COGs) in leafletjs. The COG's are in a range of projections and being displayed on a standard leafletjs web mercator projection.

I'm retrieving tiles but when placed on the map the edges are out of alignment
<img width="1376" alt="sample" src="https://user-images.githubusercontent.com/6735870/190393193-3f8cdccd-4f9b-445d-81bb-afbd6452e05f.png">

Do you think your transform and/or render library would be able to help in the situation? I normally use proj4js to do coordinate transformation.

Regards,
Rowan

**Comment by bertspaan** [comment]
(I converted this issue to a discussion! Hope you don't mind!)

I guess I don't know enough about rendering COGs in Leaflet. Are you using the Canvas API? Or WebGL?

**Reply by rowanwins** [reply]
I'm attempting to render the data onto a canvas (where in theory I'm hoping I could do some transforms/warping) which I then convert to an image which I'm placing on the map using a [L.imageOverlay](https://leafletjs.com/reference.html#imageoverlay) with the WGS84 tile bounds.

**Reply by bertspaan** [reply]
I think you're better off using proj4js. Allmaps' code is very focused on IIIF images. The transform module implements a single function from [gdaltransform](https://gdal.org/programs/gdaltransform.html) (ported to JS), and the render module focuses on finding out which part of a IIIF image is needed to render a warped map.

I'm sure there are parts of Allmaps render/transform which overlap with your use case, but I don't think it's worth it to try to use any of my modules. Besided, I'm still working on the modules' APIs, and things might change over the coming months.

I'm planning to create a Leaflet plugin for Allmaps at some point, but I'll probably have time to work on that only next year.
